airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -36,32 +36,56 @@ class KubernetesHook(BaseHook):
     """
     Creates Kubernetes API connection.
 
+    - use in cluster configuration by using ``extra__kubernetes__in_cluster`` in connection
+    - use custom configuration either by providing content of kubeconfig file via
+        ``extra__kubernetes__kube_config`` in connection
+    - use custom config by providing path to the file using ``extra__kubernetes__kube_config_path``
+    - use default config by providing no extras
+
+    .. seealso::
+        For more information about Kubernetes connection:
+        :ref:`howto/connection:kubernetes`
+
     :param conn_id: the connection to Kubernetes cluster
     :type conn_id: str
     """
 
-    def __init__(self, conn_id: str = "kubernetes_default"):
+    def __init__(
+        self, conn_id: str = "kubernetes_default", client_configuration: Optional[client.Configuration] = None
+    ):
         super().__init__()
         self.conn_id = conn_id
+        self.client_configuration = client_configuration
 
     def get_conn(self):
         """
         Returns kubernetes api session for use with requests
         """
         connection = self.get_connection(self.conn_id)
         extras = connection.extra_dejson
+
+        kubeconfig_path = extras.get("extra__kubernetes__kube_config_path")
+        kubeconfig = extras.get("extra__kubernetes__kube_config")
+
         if extras.get("extra__kubernetes__in_cluster"):
```

Comment:
Can you handle mutually exclusive configuration e.g. extra__kubernetes__kube_config + extra__kubernetes__kube_config_path at the same time?

Reply:
I understand from thecode that path overrides kubeconfig -but it would indeed be great to explain that in the docstring.

Reply:
There are 4 connection setup methods. One when the configuration is empty, but the next 3 options are mutually exclusive - `kubeconfig_path`, `kubeconfig`, `in_cluster`. 
You cannot define a configuration that contains the following sets of options:
- `in_cluster` and `kubeconfig`, 
- `in_cluster` and `kubeconfig_path`, 
- `kubeconfig` and `kubeconfig_path`.
If someone tries to use such a configuration, I think it is worth informing the user that he has a wrong configuration, because we are not able to determine the correct variant, because we only load one variant.

Reply:
@mik-laj won't this line from hook suffice?
> This hook checks for configuration options in the above order. Once an option is present it will use this configuration.

If not, can you be more precise about solution you would like to see?

Reply:
This is documentation, but I think it is also worth prohibiting misconfiguration in the code.

Wait a second. I will create the change and we will talk further.

Reply:
Can you look at it?
https://github.com/apache/airflow/pull/10453/commits/9d98d1f819c7f8d92c9c655fc87cfc828e295b14